common/dynamicconfig/file_based_client.go

```diff
@@ -248,6 +249,16 @@ func loadFile(contents []byte) (configValueMap, *LoadResult) {
 
 			cvs[i].Value = val
 			cvs[i].Constraints = convertYamlConstraints(key, cv.Constraints, precedence, lr)
+
+			// check for default value
+			if cvs[i].Constraints == (Constraints{}) {
+				hasDefaultConstraint = true
+			}
+		}
+		// add default value if not present
```

Comment:
this feels kind of weird. why do we need to insert the default values here? first of all, it doesn't work for settings with multiple constrained defaults. second, it's using more memory for something we don't actually need for looking up settings. why can't the code to retrieve settings just have logic to get the default from the registry instead of doing it here? it needs that anyway to get the default for a setting that's not preset in the file at all, right?